response_operations_ui/contexts/reporting_units.py

```diff
@@ -0,0 +1,199 @@
+            "Tel": respondent["telephone"],
+        }
+        row["account_status"] = (
+            '<span class="ons-status '
+            + _select_account_status_class(respondent["status"])
+            + '">'
+            + respondent["status"].capitalize()
+            + "</span>"
+        )
+        row["enrolment_status"] = _build_enrolment_status(respondent, ru, survey, permissions)
+        if permissions[1]:
+            row["message"] = [
+                {"name": "ru_ref", "value": ru["sampleUnitRef"]},
+                {"name": "business_id", "value": ru["id"]},
+                {"name": "business", "value": ru["name"]},
+                {"name": "survey", "value": survey["shortName"]},
+                {"name": "survey_id", "value": survey["id"]},
+                {"name": "msg_to_name", "value": row["contact_details"]["Name"]},
+                {"name": "msg_to", "value": respondent["id"]},
+            ]
+        table.append(row)
+
+    return table
+
+
+def _select_account_status_class(status):
+    if status == "SUSPENDED":
+        return "ons-status--error"
+    else:
+        return "ons-status--success"
+
+
+def _build_enrollment_status_hyperlink(respondent, ru, survey, permissions):
+    if respondent["enrolmentStatus"] == "ENABLED":
+        status_class = "ons-status--success"
+        hyperlink = url_for(
+            "reporting_unit_bp.confirm_change_enrolment_status",
+            ru_ref=ru["sampleUnitRef"],
+            ru_name=ru["name"],
+            survey_id=survey["id"],
+            survey_name=survey["shortName"],
+            respondent_id=respondent["id"],
+            respondent_first_name=respondent["firstName"],
+            respondent_last_name=respondent["lastName"],
+            business_id=ru["id"],
+            trading_as=ru["trading_as"],
+            change_flag="DISABLED",
+            tab="reporting_units",
+        )
+        hyperlink_text = "Disable"
+    elif respondent["enrolmentStatus"] == "PENDING":
+        status_class = "ons-status--info"
+        hyperlink = url_for(
+            "reporting_unit_bp.view_resend_verification", ru_ref=ru["sampleUnitRef"], party_id=respondent["id"]
+        )
+        hyperlink_text = "Re-send verification email"
+    else:
+        status_class = "ons-status--dead"
+        hyperlink = url_for(
+            "reporting_unit_bp.confirm_change_enrolment_status",
+            ru_ref=ru["sampleUnitRef"],
+            ru_name=ru["name"],
+            survey_id=survey["id"],
+            survey_name=survey["shortName"],
+            respondent_id=respondent["id"],
+            respondent_first_name=respondent["firstName"],
+            respondent_last_name=respondent["lastName"],
+            business_id=ru["id"],
+            trading_as=ru["trading_as"],
+            change_flag="ENABLED",
+            tab="reporting_units",
+        )
+        hyperlink_text = "Re-enable"
+    if not permissions[0]:
+        hyperlink = None
```

Comment:
This feels strange, you set it above, then clear it out if no permission, feels the wrong order

Reply:
Updated this to make it make more sense